opensrp-giz-malawi/src/main/java/org/smartregister/giz/repository/ReasonForDefaultingRepository.java

```diff
@@ -0,0 +1,110 @@
+        ContentValues values = new ContentValues();
+        values.put(GizConstants.JsonAssetsHelper.ID, reasonForDefaultingModel.getId());
+        values.put(GizConstants.JsonAssetsHelper.OUTREACH_DATE, reasonForDefaultingModel.getOutreachDate());
+        values.put(GizConstants.JsonAssetsHelper.FOLLOWUP_DATE, reasonForDefaultingModel.getFollowupDate());
+        values.put(GizConstants.JsonAssetsHelper.OUTREACH_DEFAULTING_REASON, reasonForDefaultingModel.getOutreachDefaultingReason());
+        values.put(GizConstants.JsonAssetsHelper.ADDITIONAL_DEFAULTING_NOTES, reasonForDefaultingModel.getAdditionalDefaultingNotes());
+        values.put(GizConstants.JsonAssetsHelper.BASE_ENTITY_ID, reasonForDefaultingModel.getBaseEntityId());
+        values.put(GizConstants.JsonAssetsHelper.EVENT_DATE, reasonForDefaultingModel.getDateCreated());
+        values.put(GizConstants.JsonAssetsHelper.OTHER_DEFAULTING_REASON, reasonForDefaultingModel.getOtherOutreachDefaultingReason());
+        return values;
+    }
+
+    public ArrayList<ReasonForDefaultingModel> readAllReasonsForDefaulting() {
+        String[] columns = {GizConstants.JsonAssetsHelper.ID, GizConstants.JsonAssetsHelper.OUTREACH_DATE,
+                GizConstants.JsonAssetsHelper.FOLLOWUP_DATE, GizConstants.JsonAssetsHelper.OUTREACH_DEFAULTING_REASON, GizConstants.JsonAssetsHelper.ADDITIONAL_DEFAULTING_NOTES,
+                GizConstants.JsonAssetsHelper.BASE_ENTITY_ID, GizConstants.JsonAssetsHelper.EVENT_DATE, GizConstants.JsonAssetsHelper.OTHER_DEFAULTING_REASON};
+        Cursor cursor = getReadableDatabase().query(TABLE_NAME, columns, null, null, null, null, null, null);
+        ArrayList<ReasonForDefaultingModel> reasonForDefaultingModels = new ArrayList<>();
+
+        try {
+            if (cursor != null && cursor.moveToFirst()) {
+                do {
+                    reasonForDefaultingModels.add(new ReasonForDefaultingModel(
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.ID)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.OUTREACH_DATE)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.FOLLOWUP_DATE)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.OUTREACH_DEFAULTING_REASON)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.ADDITIONAL_DEFAULTING_NOTES)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.BASE_ENTITY_ID)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.EVENT_DATE)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.OTHER_DEFAULTING_REASON))));
+
+                } while (cursor.moveToNext());
+            }
+
+        } catch (Exception e) {
```

Comment:
Catch a less general exception, line `SQLiteException` . Also why not use `try-with-resources`.

Reply:
Done